AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -997,6 +1028,7 @@ - (void)rangeController:(ASRangeController *)rangeController didDeleteNodes:(NSA
     [_layoutFacilitator collectionViewWillEditCellsAtIndexPaths:indexPaths batched:NO];
     [UIView performWithoutAnimation:^{
       [super deleteItemsAtIndexPaths:indexPaths];
```

Comment:
I did see the dispatch back to the main queue, but I believe we dispatch every time even if a dispatch has already occurred, So the check will end up running many times. This could be solved in the same way that coalescing is done by the range controller, having a simple Boolean flag that indicates that a scheduling has occurred but not yet executed, And so we skip scheduling in this case.